src/operator/numpy/np_nonzero_op-inl.h

```diff
@@ -43,6 +43,25 @@ namespace mxnet {
 namespace op {
 
 struct NonzeroForwardKernel {
+  // this is for cpu
+  template<int ndim>
+  MSHADOW_XINLINE static void Map(index_t i,
+                                  int64_t* out,
+                                  const index_t* idx,
+                                  const mshadow::Shape<ndim> shape) {
+    index_t prev = (i == 0) ? 0 : idx[i - 1];
+    index_t curr = idx[i];
+    if (prev != curr) {
+      mshadow::Shape<ndim> coord = mxnet_op::unravel<ndim>(i, shape);
+      for (int j = 0; j < ndim; j++) {
+        out[prev * ndim + j] = coord[j];
+      }
+    }
+  }
+};
+
+struct NonzeroForwardKernelGPU {
+  // for gpu implementation because it does not support int 64 indexing
```

Comment:
Can you point me to where the cub library is getting called ?

Reply:
In our code the cub function is called here: https://github.com/apache/incubator-mxnet/blob/master/src/operator/numpy/np_nonzero_op.cu#L71
You can also refer to https://github.com/apache/incubator-mxnet/issues/19020

Reply:
aah ... makes sense two Fcompute each for CPU and GPU comes to same kernel eventually but for GPU's FCompute there is a prior call to cub that expects an int32_t pointer that it populates and returns.